Source/ResponseSerialization.swift

```diff
@@ -515,12 +515,20 @@ extension DownloadRequest {
 }
 
 // MARK: - Empty
+/// A protocol for a type representing an empty response. Use `T.emptyValue` to get an instance.
+public protocol EmptyResponse {
+    static var emptyValue: EmptyResponse { get }
```

Comment:
In order to work around the issue you identified in the tests, where types could accidentally return some other `EmptyResponse` type, perhaps this property should have type `Self`? I just don't know if that would break the casting that needed to pull the value out.

Reply:
Guess I hadn't tried that. I'll play around with it.